### DISCUSSION TITLE
How to export dynamic component html data

### DISCUSSION THREAD

**Discussion by sateesh-fishead** [GrapesJS/grapesjs#5654] [Q&A]
I have created a custom component like product grid, on render the component am calling API and rendering dynamic html.
this is fine in admin editor mode, but while exporting data to html to render in frontend site. dynamically generated product grid html not exporting, what am missing here.

**Comment by artf** [comment]
Probably you're updating directly the DOM instead of the component model, but it's hard to tell if you don't post any reproducible demo...

**Reply by sateesh-fishead** [reply]
Hi @artf ,

I understood need to make component modal to export data instead of direct DOM manipulation, DOM manipulation can be done via JS.

I have another query,  I wanted to build custom product slider in there am using api to modify the DOM  element inside slider, How to pass dynamic API URL to access inside the script.  

Thank you.
Sateesh

**Reply by artf** [reply]
It's explained [here](https://grapesjs.com/docs/modules/Components-js.html#passing-properties-to-scripts)
